plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/EclipseProjectCreator.java

```diff
@@ -15,10 +25,57 @@
   /**
    * @return status of project creation
    */
-  static IStatus makeNewProject(AppEngineStandardProjectConfig config, IProgressMonitor monitor) {
-    IWorkspaceRoot workspaceRoot = ResourcesPlugin.getWorkspace().getRoot();
-    IProject project = workspaceRoot.getProject(config.getEclipseProjectName());
-    return Status.OK_STATUS;
+  static IStatus makeNewProject(
+      AppEngineStandardProjectConfig config, IProgressMonitor monitor,
+      final Shell shell, final IWizardContainer container) {
```

Comment:
This should be `IRunnableContext` rather than `IWizardContainer`.

But really this class should be headless, with no references to `Shell`, `IRunnableContext`, or any UI stuff.  Rather this method should be called in the context of an `IRunnableWithContext` — really it's a wrapper around the `CreateProjectOperation`.  So the method should take an `AEStandardProjectConfig`, an `IProgressMonitor`, and an `IAdaptable uiInfo` (to keep the workspace operation happy).  This class handles the orchestration of creating the project within the workspace, adding the facets, etc.  It can be called from a variety of contexts, including from a wizard, and called programmatically (e.g., from a test).